core/jobs/batch_jobs/skill_validation_jobs.py

```diff
@@ -0,0 +1,136 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation jobs for story models."""
+
+from __future__ import annotations
+
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import skill_models
+
+(skill_models, ) = models.Registry.import_models([models.NAMES.skill])
+
+
+class GetNumberOfSkillsWithInvalidRubricExplanationsJob(base_jobs.JobBase):
+    """Job that returns skills with invalid rubric explanations."""
+
+    def rubric_explanations_are_invalid(self, explanations):
+        """Returns true if the length of rubric explanations list is > 10
+        or at least one explanation exceeds 300 characters.
+
+        Returns:
+            bool. Returns true if the length of rubric explanations list is > 10
+            or at least one explanation exceeds 300 characters.
+        """
+        if len(explanations) > 10:
+            return True
+
+        for explanation in explanations:
+            if len(explanation) > 300:
+                return True
+
+        return False
+
+    def get_count_of_rubrics_with_invalid_explanations(self, skill):
+        """Returns the count of rubrics with invalid explanations in a skill.
+
+        Returns: int. The count of rubrics with invalid explanations.
+        """
+        count = 0
+        rubrics = skill[1]
+        for rubric in rubrics:
+            if self.rubric_explanations_are_invalid(rubric['explanations']):
+                count = count + 1
+        return count
+
+    def filter_skills_having_rubrics_with_invalid_explanations(self, skill):
+        """Returns True if skill has rubrics with invalid explanation.
+
+        Returns:
+            bool. Returns True if skill has rubrics with invalid explanation.
+        """
+        rubrics = skill[1]
+        for rubric in rubrics:
+            if self.rubric_explanations_are_invalid(rubric['explanations']):
+                return True
+        return False
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of skills with invalid rubric explanations.
+
+        Returns:
+            PCollection. Returns a PCollection of skills with invalid
+            rubric explanations.
+        """
+        total_skills = (
+            self.pipeline
+            | 'Get all SkillModels' >> ndb_io.GetModels(
+                skill_models.SkillModel.get_all(include_deleted=False))
+            | 'Combine skill ids and rubrics' >> beam.Map(
+                lambda skillModel: (
```

Comment:
```suggestion
                lambda skill_model: (
```

we don't use snakeCase for Python variables.

Reply:
@vojtechjelinek I think you mean that we don't use `camelCase` for Python variables? We do use `snake_case`.

Reply:
done.